woocommerce/woocommerce-android
# Issue/13163 logic to save hidden sites

Closes: #13163


### Description

Adds all the logic to: 
- Persist site selection is data store
- Update site list from site picker according to the selected sites: 

### Testing information
- Log in with an account that has multiple sites
- Click Edit Stores button
- Edit selected stores 
- Tap Save
- Navigate back to site picker and check the unselected sites are hidden
- Kill the app and navigate to site picker and check only selected sites are displayed

### The tests that have been performed

The above

### Images/gif


https://github.com/user-attachments/assets/1463ec58-0a38-4a8b-8ef5-82f360ea1e38


## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
